File: dehaze-python/app/models/schema/api_key.py
from datetime import datetime

from pydantic import BaseModel, Field


class ApiKeyCreate(BaseModel):
    name: str = Field(..., min_length=1, max_length=128, description="密钥名称")
    expiresAt: datetime | None = Field(default=None, description="过期时间")
    dailyQuota: int | None = Field(default=None, ge=1, description="日调用配额(不限制则不传或传0)")
    monthlyQuota: int | None = Field(
        default=None, ge=1, description="月调用配额(不限制则不传或传0)"
    )
    rpmLimit: int | None = Field(
        default=None, ge=1, description="每分钟请求数上限RPM(不限制则不传或传0)"
    )
    modelWhitelist: list[str] | None = Field(
        default=None, description="模型白名单(不限制或继承用户可见模型则不传或传空数组)"
    )


class ApiKeyResult(BaseModel):
    id: int = Field(description="密钥ID")
    name: str = Field(description="密钥名称")
    apiKey: str | None = Field(default=None, description="密钥明文(仅创建时返回)")
    keyPrefix: str = Field(description="密钥前缀")
    status: int = Field(description="状态(1:正常;0:禁用)")
    expiresAt: datetime | None = Field(default=None, description="过期时间")
    lastUsedAt: datetime | None = Field(default=None, description="最后使用时间")
    createTime: datetime | None = Field(default=None, description="创建时间")
    dailyQuota: int | None = Field(default=None, description="日调用配额(NULL表示不限制)")
    monthlyQuota: int | None = Field(default=None, description="月调用配额(NULL表示不限制)")
    rpmLimit: int | None = Field(default=None, description="每分钟请求数上限RPM(NULL表示不限制)")
    modelWhitelist: list[str] | None = Field(
        default=None, description="模型白名单(NULL表示继承用户可见模型)"
    )

File: dehaze-go/internal/model/dto/api_key.go
package dto

import "time"

type ApiKeyCreateRequest struct {
	Name      string     `json:"name" binding:"required"`
	ExpiresAt *time.Time `json:"expiresAt"`
	// 治理参数：不传即不限制（python ApiKeyCreate 的 ge=1 校验同口径）
	DailyQuota   *int64 `json:"dailyQuota" binding:"omitempty,min=1"`
	MonthlyQuota *int64 `json:"monthlyQuota" binding:"omitempty,min=1"`
	RpmLimit     *int   `json:"rpmLimit" binding:"omitempty,min=1"`
	// 模型白名单：不传或空数组 = 继承用户可见模型（python ApiKeyCreate.modelWhitelist 同口径）
	ModelWhitelist []string `json:"modelWhitelist"`
}

type ApiKeyResult struct {
	ID         int64      `json:"id"`
	Name       string     `json:"name"`
	ApiKey     string     `json:"apiKey,omitempty"`
	KeyPrefix  string     `json:"keyPrefix"`
	Status     int8       `json:"status"`
	ExpiresAt  *time.Time `json:"expiresAt"`
	LastUsedAt *time.Time `json:"lastUsedAt"`
	CreateTime time.Time  `json:"createTime"`
	// 治理参数回传（python ApiKeyResult 同形态；NULL 表示不限制，null 字段整键省略）
	DailyQuota     *int64   `json:"dailyQuota"`
	MonthlyQuota   *int64   `json:"monthlyQuota"`
	RpmLimit       *int     `json:"rpmLimit"`
	ModelWhitelist []string `json:"modelWhitelist,omitempty"`
}
